pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -319,6 +327,7 @@ def newpkgcb(pkgId, name, arch):
 
             repomd_path = result.path
             repomd = cr.Repomd(repomd_path)
+            repomd_revision = "".join(re.findall(r'\d', repomd.revision))
```

Comment:
Not the best solution,
For CentOS 8 AppStream, the revision is not a timestamp, it is: `8.1.1911`
http://mirror.linux.duke.edu/pub/centos/8/AppStream/x86_64/kickstart/repodata/repomd.xml
From a quick search about revisions, I saw that sometimes it could be empty:
https://pulp.plan.io/projects/pulp_rpm/repository/9/revisions/54c8837a8e5bd80b9df16689b0cec1c3410e8a61
https://bugzilla.redhat.com/show_bug.cgi?id=1332922

Reply:
I'm not sure we should do that.
Imagine the next revision will be 9.0.0 and optimization will happen when it should not have. Without knowing their (or any other) system of versioning for a revision, we can't be sure. And for correctness sake, we should not optimize in such cases.
I think if the revision is not int, the only safe thing that I see is to save the revision as is and compare such values for equality. And only if both (saved revision and incoming revision) are converted to integers we can compare.

Thoughts?

Reply:
I agree.

Reply:
saving as is means I have to change `last_sync_revision_number` to `CharField`, but I don't get why we should compare only if both are numbers.  But I agree we should not optimize when revision is empty

Reply:
I suggested to compare strings for equality only. If both are numbers, than we can see if one is older than the other.

Reply:
There are a number of places in rpm-land with this kind of unreliable version-format. We have code in several projects that "try to do the right thing" in the face of empty/timestamp/int/x.y.z/iso-date-format/arbitrary-string 'versions'. I would suggest we either (re)use one of these 'try to be as correct as possible' implementations, or we optimize only when a straight-string-compare returns 'equal'. I suppose we *could* do a halfway "if casting to int works do an int-compare, else check for string-equality, else skip optimizing" - but if we're going to do that, it's not that big a leap to just add the rest of logic used in the other version-comparing code bases, and be able to (for example) recognize X.Y.Z and that 8.2 is 'less than' 8.10 (which fails if you do either a string-compare or a 'try them as floats' compare)